pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -581,9 +582,11 @@ func (c *Client) GetVPC(ctx context.Context, vpcID string) (*vpcv1.VPC, error) {
 		}
 
 		if vpc, detailedResponse, err := c.vpcAPI.GetVPC(c.vpcAPI.NewGetVPCOptions(vpcID)); err != nil {
-			if detailedResponse.GetStatusCode() != http.StatusNotFound {
-				return nil, err
+			if detailedResponse != nil && detailedResponse.GetStatusCode() != http.StatusNotFound {
+				logrus.Warnf("Failure collecting VPC %s in %s - %q", vpcID, *region.Name, err)
+				continue
 			}
+			logrus.Warnf("Response status unknown while checking VPC %s in %s region", vpcID, *region.Name)
```

Comment:
The status code could still be `NotFound` but you're logging that it's unknown. Maybe it's better to include `detailedResponse` in the log message?

Reply:
Okay, let me rework the logic/logs some for these two blocks.

Reply:
Working on an update to reconfigure these blocks....

Reply:
Made updates, retested to confirm this still addresses bad regional API's, instead of the go panic.
```
DEBUG   Reusing previously-fetched Common Manifests 
DEBUG Generating Terraform Variables...            
WARNING Failure collecting VPCs in us-east: "Get \"https://<bad-regional-endpoint-redacted>.cloud.ibm.com/v1/vpcs?generation=2&version=2023-08-08\": dial tcp: lookup <bad-regional-endpoint-redacted>.cloud.ibm.com on 127.0.0.53:53: no such host" 
DEBUG Obtaining RHCOS image file from 'https://rhcos.mirror.openshift.com/...
```